src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -425,6 +426,14 @@ private void StartSend(HttpClient client, HttpRequestMessage message)
 
         private async Task SendAsync(HttpClient client, HttpRequestMessage message)
         {
+            // In C-Core the call credential auth metadata is only applied if the channel is secure
+            // The equivalent in grpc-dotnet is only applying metadata if HttpClient is using TLS
+            // HttpClient scheme will be HTTP if it is using H2C (HTTP2 without TLS)
+            if (Options.Credentials != null && client.BaseAddress.Scheme == Uri.UriSchemeHttps)
```

Comment:
This seems good enough from the security perspective, but it might not be exactly what C-core does.
I'm actually not sure if C-core invokes the AuthInterceptor callback always and then strips away the metadata entries generated by the callback for insecure channels or if it just skips the call to the interceptor entirely.

If C-core always invokes the auth interceptor callback (even for insecure channels), we should probably do the same (and then not add the metadata to the call and perhaps print a debug log message).

Reply:
I tested and the interceptor isn't run at all for unsecure.